Simon-Initiative/oli-torus
# [FEATURE] Pagination policies, automation of page display

This PR adds the ability for an author to control how content breaks are paginated.  It also allows feedbacks within paginated activities to automate the "reveal" of other groups of content.  This combination of features realizes the legacy "Branching assessment" feature. 

Some of the key aspects of the changes here include:
1. Creation of a more strongly typed client-side "Eventing" subsystem.  This is used for the new `ShowPage` event and the previous events related to surveys. 
2. Three modes for paginated content: User controlled pagination (what existed before), user controlled 'reveal', and programmatic (aka 'automated') reveal where something else on the page sends a message to the pagination controls to trigger the reveal. 
3. Cleaned up the DeliveryElement webcomponent interface, adding a `context` required parameter that now contains the previous `graded`, `survey_id`, `group_id`, `user_id`, etc.  All attributes are now required as opposed to allowing `undefined`, replaced with the value of `null` for special meaning for group and survey.
4. UI changes to allow paginated groups to configure their pagination mode.
5. UI changes to allow activities within paginated groups to specify the automated reveal of another page. 
6. Server side changes to pass this `show_page` parameter back to the client upon evaluation, and client-side handling to dispatch this event.
7. Leverage the "page attempt" extrinsic state to preserve the state of reveal style paginated groups.  This allows a student to reveal items in a page, navigate away and return to the page with that state preserved. 

## To Test this out

1. Create a practice page.
2. In the page, create three groups.  Add content breaks and content to all three groups.
3. Verify that only after content breaks appear that the pagination mode UI appears.
4. Verify that only after content breaks appear that the Feedback option for automating page display appears.
5. Verify that if all content breaks are deleted the pagination mode and the automating page display options disappear.
6. Restore all breaks, set the pagination mode for each of the three groups to each of the three pagination modes.
7. In the one with "automated reveal" as the option, set feedbacks some some of the activities present in that group to reveal other page groups in that group. 
8. Publish and enter the course section as a student.  Verify that regular pagination works, manual reveal, and the automated reveal via feedbacks received works.
9. Change the page type to be "graded" then re-publish.  Access the page as a student as a graded page and verify that everything works *EXCEPT* there is no way to trigger the feedback received page display commands, since graded page activities are not evaluated until the entire page is submitted. 
10. Submit the entire page and verify in the "Review Attempt" mode that all questions are displayed and were scored.
